core/native/src/files/PathsNative.kt

```diff
@@ -18,13 +18,61 @@ import platform.posix.*
 public actual class Path internal constructor(
     internal val path: String,
     @Suppress("UNUSED_PARAMETER") any: Any?
-)
+) {
+    public actual val parent: Path?
+        get() {
+            when {
+                path.isBlank() -> return null
+                !path.contains(separator) -> return null
+            }
+            val parentName = dirnameImpl(path)
+            return when {
+                parentName.isBlank() -> return null
+                parentName == path -> return null
+                else -> Path(parentName)
+            }
+        }
+
+    public actual override fun toString(): String = path
+    actual override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other !is Path) return false
+
+        return path == other.path
+    }
+
+    actual override fun hashCode(): Int {
+        return path.hashCode()
+    }
+
+    public actual companion object {
+        public actual val separator: Char = '/'
```

Comment:
Windows supports both `/` and `\` as path separators. As `mingw` brings some GNU goodies to Windows, I decided to stick with `/`.

If there's a reason to choose `\` instead of `/` (like there are cases when Windows can't correctly handle it), I would definitely change the separator to `\`.